File: Extra/leetcode/944. Delete Columns to Make Sorted/Solution.java
class Solution {
  public int minDeletionSize(String[] strs) {
    int len = strs[0].length();
    int res = 0;

    for (int i = 0; i < len; i++) {
      for (int j = 1; j < strs.length; j++) {
        if (strs[j].charAt(i) < strs[j - 1].charAt(i)) {
          res++;
          break;
        }
      }
    }

    return res;
  }
}

File: Extra/leetcode/944. Delete Columns to Make Sorted/Solution.scala
object Solution {
  def minDeletionSize(strs: Array[String]): Int = {

    def aux(strs:Array[String], cont:Int, indexStart:Int, indexEnd:Int, charIndex:Int):Int = {
      (indexStart == indexEnd) match {
        case true => {
          charIndex == strs.length-1 match {
            case true => cont
            case false => aux(strs, cont+1, 0, strs.length-1, charIndex+1)  
          }
        }
        case false => {
          strs(indexStart).charAt(charIndex) < strs(indexEnd).charAt(charIndex) match {
            case true => aux(strs, cont, indexStart, index)
          }
        } 
      }
    }

    /*
    strs(0).charAt(i) < strs(strs.length-1).charAt(i) 
    case true => strs(0).charAt(i) < strs(strs.length-2).charAt(i)
    case false => inc cont, pass to next index
    */

  }
}